strawberryfields/ops.py

```diff
@@ -1730,7 +1732,17 @@ class BSgate(Gate):
     """
     ns = 2
 
-    def __init__(self, theta=np.pi / 4, phi=0.0):
+    def __init__(self, theta=np.pi / 4, phi=None):
+        if (np.iscomplex([theta, phi])).any():
+            raise ValueError("The arguments of BSgate(theta, phi) cannot be complex")
+        if phi is None:
+            phi = 0.0
+            # TODO: remove warning in the new release
+            warnings.warn(
+                f"""Warning: since strawberryfields version v0.15.0,
+            BSgate(theta, phi) takes two angles rather than the transmission and
+            reflection coefficients. Using the arguments theta={theta} and phi=0.0"""
+            )
```

Comment:
It seems to me that the UI for the BSgate and the S2gate did not change. It's only that after taking the real parameters, internally the complex argument was being used.

This can be checked by having a look at the history (clicking on the line number and choosing to view the git blame).

Link to the BSgate: https://github.com/XanaduAI/strawberryfields/blob/3c7b2a7bec0e05cf2133643bc4f63643db2e5a0e/strawberryfields/ops.py#L1738

Reply:
So, the BSgate now takes 2 angles rather than t,r (one of which could be complex), so there has been some change.

I was convinced I had changed the S2gate, but you're probably right, I've changed so many things lol 😄

For the other operations, for some reason I was convinced I had not modified them in ops.py 🤦‍♂️💆‍♂️
Thank you for the reminder, I'll just shift the errors/warnings there

Reply:
Was that the case really with the BSgate? :thinking: It seems to me that we were deriving t,r using
```python
t = pf.cos(self.p[0])
r = pf.sin(self.p[0]) * pf.exp(1j * self.p[1])
```
before they got passed to
```python
t = par_evaluate(t)
r = evaluate_complex_parameter(r, self.p[1])
backend.beamsplitter(t, r, *reg)
```
Sure thing! :slightly_smiling_face: Yes, it's not so easy to look at the changes after a while :sweat_smile:

Reply:
Can verify, the `BSgate` hasn't changed, it has been `BSgate(theta, phi)` from the beginning 🙂 

What _has_ changed is the backend API; the backend call used to be `BaseBackend.beamsplitter(t, r)`, requiring the frontend to convert the input theta, phi into transmission and reflection. But the UI has not changed.